plugins/module_utils/users.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2024, Jeff Kala (@jeffkala)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+from ansible_collections.networktocode.nautobot.plugins.module_utils.utils import (
+    NautobotModule,
+    ENDPOINT_NAME_MAPPING,
+)
+
+
+NB_USERS = "users"
+NB_ADMIN_GROUP = "groups"
+NB_OBJECT_PERMISSION = "permissions"
+
+
+class NautobotUsersModule(NautobotModule):
+    def run(self):
+        """
+        This function should have all necessary code for endpoints within the application
+        to create/update/delete the endpoint objects
+        Supported endpoints:
+        users
+        admin_groups
+        object_permissions
```

Comment:
I like the `admin_*` option, originally why I added admin to the group was it appeared it would collide with tenant group which in places looked like it was just called group.  I may have misinterpreted it though.
Will go with the `admin_*` rename.